web/pandas/pdeps/0003-small-data-visualization-consolidation.md

```diff
@@ -0,0 +1,158 @@
+# PDEP-3: Small Data Visualization Consolidation
+
+- Created: 13 August 2022
+- Status: Draft
+- Discussion: [#48080](https://github.com/pandas-dev/pandas/pull/48080)
+- Author: [JHM Darbyshire](https://github.com/attack68)
+- Revision: 1
+
+## Abstract
+
+Developing consistent and flexible output methods for small data,
+which are well documented and decoupled from `DataFrame` methods.
+
+## Motivation and Scope
+
+Large data is not in scope for this PDEP. Large Data is classed as that which
+would not be expected to print, or be wire-transferred for example, with LaTeX,
+HTML or JSON.
```

Comment:
Excel somewhat muddies the water here; but I don't see any issue with this PDEP if this were to be removed. Trying to convert a 100k row DataFrame to LaTeX is a bad idea, but if a user requested such an operation, pandas would give it a go right?

Reply:
There is a default pandas option for Styler to render 2**18 = 260k elements without data trimming. This was chosen due to some browser CSS styling constraints and the time it takes (about 4 seconds in Jupyter). If a user wanted to they could alter this setting and render as many elements as they wanted. Styler output for LaTeX, HTML, String and potentially JSON is just a string output.

There have been one or two issues that arose for Styler when dealing with large data, and that needed separate handling. I included this line just to explicitly avoid the need for those special considerations.

Reply:
I see - thanks! Am I correct in saying that such trimming doesn't apply to the already existing `Styler.to_excel` and won't apply to the proposed `Styler.to_json`? I'm certainly good with explicit saying something to the effect as you mentioned above, but it does seem to me part of this PDEP potentially involves working with big data.